File: two pointers/palindrome/palindrome.py
import re


def is_palindrome(str):
  def clean_string(str):
    str = str.lower()
    return re.sub(r'\W+','',str)
  
  str = clean_string(str)
  leftPtr = 0
  rightPtr = len(str) - 1
  
  while leftPtr < rightPtr:
    if str[leftPtr] != str[rightPtr]:
      return False
    else:
      leftPtr = leftPtr + 1
      rightPtr = rightPtr - 1
  return True

print(is_palindrome('A man, a plan, a canal: Panama'))

File: two pointers/palindrome/palindrome.js
function cleanString(str){
  str = str.replace(/[\W+_]/g,'');
  return str.toLowerCase();
}

function isPalindrome(str){
  str = cleanString(str)
  let leftPtr = 0;
  let rightPtr = str.length - 1; 

  while(leftPtr < rightPtr){
    if(str[leftPtr] != str[rightPtr]){
      return false;
    }else{
      leftPtr++;
      rightPtr--;
    }
  }
  return true;
}

console.log(isPalindrome('A man, a plan, a canal: Panama'))